### ISSUE TITLE
Cocoapods 1.15.1 no longer includes xcdatamodel files

### ISSUE THREAD

**Issue by foux** [CocoaPods/CocoaPods#12264]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
pod install

## What did you expect to happen?
My pod should be installed, including `xcdatamodel` files

## What happened instead?
My pod is correctly installed, except for the xcdatamodel files

### Stack

```
   CocoaPods : 1.15.1
        Ruby : ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [arm64-darwin23]
    RubyGems : 3.5.4
        Host : macOS 14.3 (23D56)
       Xcode : 15.2 (15C500b)
         Git : git version 2.39.3 (Apple Git-145)
Ruby lib dir : /opt/homebrew/Cellar/ruby/3.3.0/lib
Repositories : internal - git - https://gitlab.limberapps.com/iOS/internal-podspecs.git @ bce162f1f6f3ee5fc66521ee051ce3c89685b285

               limberapps-ios-internal-podspecs - git - https://gitlab.limberapps.com/iOS/internal-podspecs.git @ 546e4c0134d621e2ef8d1b343fbe97af8dda6074

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/foux/GIT/Limberapps/salesapps-my/vendor/bundle/ruby/3.3.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
platform :ios, '13.0'

source 'https://cdn.cocoapods.org/'
source 'https://gitlab.limberapps.com/iOS/internal-podspecs.git'
 
use_frameworks!
inhibit_all_warnings!
 
target 'Salesapps' do
    pod 'raygun4apple'
    pod 'LimberAppMaster', '= 7.15.2'

end

post_install do |installer_representation|
    `rm -rf Pods/Headers/Private`
    `find Pods -regex 'Pods\/.*\.modulemap' -print0 | xargs -0 sed -i '' 's/private header.*//'`

    installer_representation.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            #desactive 
            config.build_settings['EXPANDED_CODE_SIGN_IDENTITY'] = ""
            config.build_settings['CODE_SIGNING_REQUIRED'] = "NO"
            config.build_settings['CODE_SIGNING_ALLOWED'] = "NO"
            config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
            if config.name == 'Preprod'
                config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', 'PREPROD=1']
            end
        end
    end
    
    target = installer_representation.pods_project.targets.find{|t| t.to_s == "MagicalRecord"}
    target.build_configurations.each do |config|
        s = config.build_settings['GCC_PREPROCESSOR_DEFINITIONS']
        s = [ '$(inherited)' ] if s == nil;
        s.push('MR_LOGGING_DISABLED=1') if config.to_s == "Debug";
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] = s
    end
end
```

## Additional infos
No issues on 1.15.0

**Comment by khoogheem** [comment]
I am seeing all sorts of issues from Modules not found to Build script issues with this version..
Need a reverted Brew formula.. its broken all my projects

**Comment by foux** [comment]
> I am seeing all sorts of issues from Modules not found to Build script issues with this version.. Need a reverted Brew formula.. its broken all my projects

Simply use a gemfile and you can revert as you like

**Comment by khoogheem** [comment]
can't use the gem versions due to the ruby version it wants.. which is why we switched to brew...
Until they can fix the issues with 15.1 taking it down from Brew would help as It is a complete nightmare to try to use older version in Brew

**Comment by monfresh** [comment]
If you can't install the gem, that means you're trying to install it using the system Ruby (the version of Ruby that came preinstalled on your Mac). I don't recommend that, and here are a few reasons [Why You Shouldn't Use the System Ruby to Install Gems on a Mac](https://www.moncefbelyamani.com/why-you-shouldn-t-use-the-system-ruby-to-install-gems-on-a-mac/). 

Instead, I recommend using a version manager to install a separate and newer version of Ruby that doesn't interfere with the system Ruby, and that will allow you to easily install Cocoapods as a gem, without using `sudo`.

Here's a [step-by-step guide for setting up a proper Ruby environment on a Mac](https://www.moncefbelyamani.com/how-to-install-xcode-homebrew-git-rvm-ruby-on-mac/).

**Comment by khoogheem** [comment]
That is great and all at this point I can't update the ruby on device and revert back to using Gems.. I rely on Brew for this and that version is Busted...

**Comment by pj4533** [comment]
Agreed...using brew to install on our CI machine (Xcode Cloud), so this broke our builds since 1.15.1 is busted.

**Comment by khoogheem** [comment]
I was able to take a copy of the `/opt/homebrew/Cellar/cocoapods/1.15.0` from a different machine and relink that copy and I am back up.. but what a PIA!!!

**Comment by paulb777** [comment]
1.15.2 is now available. It should fix this issue. Can anyone confirm?

**Comment by pj4533** [comment]
Doesn't look like 1.15.2 is on Homebrew (yet)?

**Comment by foux** [comment]
> 1.15.2 is now available. It should fix this issue. Can anyone confirm?

Working on my end, my xcdatamodel version is back here!

**Comment by monfresh** [comment]
As you have experienced, the problem with installing Cocoapods with Homebrew is that you can't easily choose which version you want, or easily downgrade. That's why I recommend taking the time to learn how to set up a Ruby development environment with a version manager such as chruby/ruby-install, asdf, or rbenv, all of which can be installed with Homebrew. And then you can use that version manager to install a specific version of Ruby, with a command such as `ruby-install 3.2.3`.

**Comment by pj4533** [comment]
Yeah confirmed, 1.15.2 works great locally, managed with gems. Still busted with brew (for our CI on Xcode Cloud)...but assuming it'll get there soon?
